configuration/configuration.go

```diff
@@ -834,6 +867,54 @@ func (c *Client) setFieldValue(section parser.Section, sectionName string, field
 		}
 		return nil
 	}
+	if fieldName == "ExternalCheck" {
+		enabled := field.Bool()
+		if section == parser.Global {
+			if enabled {
+				d := &types.Enabled{}
+				if err := p.Set(section, sectionName, "external-check", d); err != nil {
+					return err
+				}
+			} else {
+				_ = p.Delete(section, sectionName, "external-check")
+			}
+		} else if section == parser.Backends || section == parser.Defaults {
+			d := &types.SimpleOption{
+				NoOption: !enabled,
+			}
+			if err := p.Set(section, sectionName, "option external-check", d); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	if fieldName == "ExternalCheckPath" {
+		if valueIsNil(field) {
+			return nil
```

Comment:
Here you have to delete the value so call p.Set(section, sectionName, "option external-check", nil)

Reply:
I think "set it to nil" is unnecessary.

Reply:
It is, try adding to https://github.com/haproxytech/client-native/blob/master/configuration/backend_test.go#L169 following lines:

```
ExternalCheck:        "enabled",
ExternalCheckCommand: "/bin/false",
ExternalCheckPath:    "/bin",
```

It will create a backend with those parameters, and if you try to edit it without those parameters they won't get deleted. the valueIsNil() function checks wheter the value of the field is "nil value", in this case wheter it is "", when the value is "" (this means not set) we need to delete the value of that property, and we can do that by sending ```nil``` as value.

Reply:
I see. Done.

Reply:
I want you to add here:

```
if valueIsNil(field) {
			if err := p.Set(section, sectionName, "external-check path", nil); err != nil {
				return err
			}
			return nil
		}
```

Reply:
I believe following piece is equivalent to your suggestion:
```
	if fieldName == "ExternalCheckPath" {
		pExtPath := &types.ExternalCheckPath{}
		if valueIsNil(field) {
			pExtPath = nil
		} else {
			pExtPath.Path = field.Interface().(string)
		}
		if err := p.Set(section, sectionName, "external-check path", pExtPath); err != nil {
			return err
		}
		return nil
	}

```